backend/src/routes/api/nim-serving/index.ts

```diff
@@ -1,40 +1,61 @@
 import { KubeFastifyInstance, OauthFastifyRequest } from '../../../types';
 import { createCustomError } from '../../../utils/requestUtils';
 import { logRequestDetails } from '../../../utils/fileUtils';
-
-const secretNames = ['nvidia-nim-access', 'nvidia-nim-image-pull'];
-const configMapName = 'nvidia-nim-images-data';
+import { getNIMAccount } from '../integrations/nim/nimUtils';
+import { secureAdminRoute } from '../../../utils/route-security';
 
 export default async (fastify: KubeFastifyInstance): Promise<void> => {
+  const resourceMap: Record<string, { type: 'Secret' | 'ConfigMap'; path: string[] }> = {
+    apiKeySecret: { type: 'Secret', path: ['spec', 'apiKeySecret', 'name'] },
+    nimPullSecret: { type: 'Secret', path: ['status', 'nimPullSecret', 'name'] },
+    nimConfig: { type: 'ConfigMap', path: ['status', 'nimConfig', 'name'] },
+  };
+
   fastify.get(
     '/:nimResource',
-    async (
-      request: OauthFastifyRequest<{
-        Params: { nimResource: string };
-      }>,
-    ) => {
-      logRequestDetails(fastify, request);
-      const { nimResource } = request.params;
-      const { coreV1Api, namespace } = fastify.kube;
+    secureAdminRoute(fastify)(
+      async (request: OauthFastifyRequest<{ Params: { nimResource: string } }>) => {
+        logRequestDetails(fastify, request);
+        const { nimResource } = request.params;
+        const { coreV1Api, namespace } = fastify.kube;
 
-      if (secretNames.includes(nimResource)) {
-        try {
-          return await coreV1Api.readNamespacedSecret(nimResource, namespace);
-        } catch (e) {
-          fastify.log.error(`Failed to fetch secret ${nimResource}: ${e.message}`);
-          throw createCustomError('Not found', 'Secret not found', 404);
+        // Fetch the Account CR to determine the actual resource name dynamically
+        const account = await getNIMAccount(fastify, namespace);
+        if (!account) {
+          throw createCustomError('Not found', 'NIM account not found', 404);
+        }
+
+        const resourceInfo = resourceMap[nimResource];
+        if (!resourceInfo) {
+          throw createCustomError('Not found', `Invalid resource type: ${nimResource}`, 404);
+        }
+
+        // Get the resource name
+        const resourceName = resourceInfo.path.reduce((acc, key) => {
+          if (acc && key in acc) {
+            return acc[key as keyof typeof acc];
+          }
+          return undefined;
+        }, account as Record<string, any> | undefined);
```

Comment:
The backend linter isn't as strict, but I'm still cautious seeing `as`. Also `resourceName` is being inferred as `any` still. Also the accumulator has multiple types since it's originally an obj but in the end it should be a string. I'm not sure